tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc

```diff
@@ -523,11 +582,81 @@ StatusOr<AutotuneResult> CudnnConvAlgorithmPicker::PickBestAlgorithmNoCache(
   return InternalError(
       "All algorithms tried for convolution %s failed.  Falling back to "
       "default algorithm.",
-      instr->ToString());
+      instr.ToString());
+}
+
+StatusOr<tensorflow::AutotuneResult>
+GpuConvAlgorithmPicker::PickBestAlgorithmNoCacheRocm(
+    const HloCustomCallInstruction& instr, se::DeviceMemoryAllocator* allocator,
+    se::Stream* stream) {
+  XLA_SCOPED_LOGGING_TIMER(absl::StrCat(
+      "GpuConvAlgorithmPicker::PickBestAlgorithmImpl for ", instr.ToString()));
+
+  const auto device_ordinal = stream_exec_->device_ordinal();
+  std::vector<se::DeviceMemoryBase> operand_buffers;
+
+  ScratchAllocator input_output_allocator(device_ordinal, allocator);
+  const auto initialize_buffer = [stream](DeviceMemoryBase buffer) {
+    // Although we don't have evidence this matters, zero out the buffers
+    // before autotuning.  It's conceivable that using uninitialized memory as
+    // the inputs might affect performance if e.g. the inputs contain
+    // denormals, and this is easy enough.
+    stream->ThenMemZero(&buffer, buffer.size());
+  };
+
+  // Allocate space for the input, filter, and output of the convolution.  We
+  // use a ScratchAllocator for this instead of calling allocator_ directly so
+  // that our allocations don't leak.
+  for (const auto* operand : instr.operands()) {
+    TF_ASSIGN_OR_RETURN(auto buffer,
+                        input_output_allocator.AllocateBytes(
+                            ShapeUtil::ByteSizeOf(operand->shape())));
+    initialize_buffer(buffer);
+    operand_buffers.push_back(buffer);
+  }
+
+  TF_ASSIGN_OR_RETURN(
+      auto result_buffer,
+      input_output_allocator.AllocateBytes(
+          ShapeUtil::ByteSizeOf(instr.shape().tuple_shapes(0))));
+  initialize_buffer(result_buffer);
+
+  ScratchAllocator scratch_allocator(device_ordinal, allocator);
+  se::dnn::ProfileResult profile_result;
+  VLOG(3) << "Auto-tuning for " << instr.ToString();
+  RunConvOptions options;
+  options.profile_result = &profile_result;
+
+  // ROCm: Set the overriding algorithm to empty to remind cudnn_conv_runner
+  // that the AlgorithmConfig in running convolution needs to be empty
+  options.algo_override = se::dnn::AlgorithmDesc();
+
+  bool launch_ok =
+      RunCudnnConv(&instr, absl::MakeSpan(operand_buffers), result_buffer,
```

Comment:
Why is ROCM implementation calling `RunCudnnConv`?

Reply:
`cudnn_conv_runner` actually works for both ROCm and CUDA out of the box, and also some other HLO opt passes with `cudnn` prefix.

In the past history upstreaming ROCm-specific logic we always had to spend quite large amount proposing naming / terminology changes and causing the whole process be stalled so we opted not to do so here.

Reply:
Yeah, naming is hard.

In this case, could we simply drop `Cudnn` from the method, and probably the whole class? Just `RunConv`? @timshen91 Anything against this?

Reply:
Also it probably would need to happen in a separate PR.

Reply:
Yes, if we are to decide the renaming to happen such that all `cudnn` are dropped, I'd prefer it to be handled al-together in a separate PR after this one.

Reply:
@timshen91 is OK with it, and I think this is necessary - would you be able to do it? I think this needs to happen before this PR, as previously the runner was indeed only used for CUDA.

Reply:
@cheshire Yes I can do it. Though I don't feel like there is a strict sequence of when renaming should happen." After all, we already have a couple of `Cudnn` related components (`CudnnConvRewriter`, `CukdnnConvPaddingLegalization`) shared by both ROCM and CUDA right now in the code base. Renaming it al-together makes it easier for us as well as for you as reviewer.